graphql_compiler/tests/test_compiler.py

```diff
@@ -4181,6 +4182,93 @@ def test_fold_on_two_output_variables(self):
         compare_input_metadata(self, test_data.expected_input_metadata,
                                result.input_metadata)
 
+    def test_fold_on_two_output_variables_mssql(self):
```

Comment:
I think either the MSSQL tests should be broken out into their own `unittest.TestCase` or have their check incorporated into `check_test_data`. Seems odd to have separated tests for the same input within `CompilerTests` when all the other backends are incorporated into a single test per input.